### PR TITLE
Rename isTerminalMode to isTerminal and move the RT URL to a dedicated variable in FileTransferDetails

### PR DESCRIPTION
**Pull request by Or-Geva**
- [x] All [tests](https://github.com/jfrog/jfrog-client-go#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] All [static analysis checks](https://github.com/jfrog/jfrog-client-go/actions/workflows/analysis.yml) passed.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----
Related to:
*  https://github.com/jfrog/jfrog-cli/pull/1564
* https://github.com/jfrog/jfrog-cli-core/pull/411

### CODE REVIEW

**Review by eyalbe4 — CHANGES_REQUESTED** [review]
Since jfrog-client-go doesn't actually display the files, let's rephrase the PR title.

File: artifactory/services/download.go [hunk 1]
```diff
@@ -520,7 +521,7 @@ func (ds *DownloadService) createFileHandlerFunc(downloadParams DownloadParams,
 				return e
 			}
 			if downloadParams.IsSymlink() {
-				if isSymlink, e := ds.createSymlinkIfNeeded(downloadPath, localPath, localFileName, logMsgPrefix, downloadData, successCounters, threadId, downloadParams); isSymlink {
+				if isSymlink, e := ds.createSymlinkIfNeeded( ds.GetArtifactoryDetails().GetUrl(), localPath, localFileName, logMsgPrefix, downloadData, successCounters, threadId, downloadParams); isSymlink {
```
Comment (eyalbe4) [inline artifactory/services/download.go:524]: Redundant white-space - 
```
ds.createSymlinkIfNeeded( ds. --> ds.createSymlinkIfNeeded(ds.
```

File: artifactory/services/upload.go [hunk 2]
```diff
@@ -727,7 +727,7 @@ func (us *UploadService) CreateUploadAsZipFunc(uploadResult *utils.Result, targe
 		var saveFilesPathsFunc func(sourcePath string) error
 		if us.saveSummary {
 			saveFilesPathsFunc = func(localPath string) error {
-				return us.resultsManager.addNotFinalResult(localPath, targetUrl)
+				return us.resultsManager.addNotFinalResult(localPath, targetPath, us.ArtDetails.GetUrl())
```
Comment (eyalbe4) [inline artifactory/services/upload.go:730]: addNotFinalResult --> addNonFinalResult

File: utils/log/logger.go [hunk 3]
```diff
@@ -18,9 +18,9 @@ type LogFormat string
 // Used for coloring sections of the log message. For example log.Format.Path("...")
 var Format LogFormat
 
-// Determines whether the terminal is available. This variable should not be accessed directly,
-// but through the 'isTerminalMode' function.
-var terminalMode *bool
+// Determines whether the terminal is available. This variable should not be accessed directly (besides  tests),,
+// but through the 'isTerminal' function.
+var TerminalMode *bool
```
Comment (eyalbe4) [inline utils/log/logger.go:23]: What's the reason for making this variable public?

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

File: utils/log/logger.go (hunk 3 above)
Comment (Or-Geva) [reply utils/log/logger.go:23]: We need to be able to change the terminal environment because tests may run in a non-terminal environment.

**Pull request merged**